Use Postman API to find collection & environment in Slack command

wmedali opened the pull request:
Update the htmlextra to new version so no template files are needed.
Examples with postman echo collection that includes already test scripts
update of readme screenshots

wmedali committed:
Bump newman and htmlextra

add working server for tunnel command
delete template files

wmedali committed:
Use Postman API instead of json files

wmedali committed:
Update README

DannyDainton commented on README.md:
```diff
@@ -24,18 +24,9 @@ I've managed to workaround this and use my local machine as the host, by using t
+The `basic-newman-slack-bot` will search in your Postman workspace for the collection and the environment to execute tests. All you need is a valid API key. You can get your key from the [integrations dashboard](https://go.postman.co/integrations/services/pm_pro_api).
 
-These files will _tell_ `newman` where to make the requests too. We'll be using the [Restful-Booker API](https://restful-booker.herokuapp.com/), this is a publicly available set of endpoints, that I had no control over so it might be worth doing a quick check first, just to know that the API is alive....or you might see a lot of test failures.
+We'll be using the [Postman echo](https://docs.postman-echo.com/?version=latest), this is a publicly available Collection released by Postman team. It icnludes many examples in test scripts. 
```
> It icnludes many examples in test scripts.

DannyDainton commented on app.js:
```diff
@@ -8,10 +8,14 @@ const path       = require('path')
+app.use(bodyParser.json({ limit: "50mb" }));
 app.use(express.static('reports'));
 
+const apikey = 'PUT YOUR API KEY HERE'
```
Not sure I like this - and not if there potential for it to be hardcoded and stored in some public repo. Can this be passed in via a system environment variable etc?

wmedali replied:
you are right.  I added a new commit to handle api key from environment file

DannyDainton replied:
Nice one. At least that way it can be excluded from the repo easier :)

Commit:
Load API key from environment file

add dotenv dependency and load api key from environment file.
Update doc as well

DannyDainton commented:
I'll bring this down locally and have a look through it in slow time - not sure how much time I'm going to get but I will get to it as soon as I can.

Really appreciate you submitting it.

wmedali commented:
Hello @DannyDainton I'd like to drop another pr to send directly the report file to slack channel and react on message depending on test results.
something like valid check emoji if tests pass or X emoji if there are any failures

I thought it would be great if you can check this pr before I send another one ^_^

DannyDainton commented:
So sorry @wmedali - I completely forgot to look at this. I will take a look today 😢

wmedali commented:
Hey @DannyDainton. Any chance to get this pr completed ? 😂

DannyDainton commented:
I did but I couldn't get it working correctly with my Slack workspace when using the Postman API - I then ran out of time :( 

I have it working now - It looks like some of the copy on the Slack message has been changed, not sure about that will need to look when I have time. The readme needs to be tidied up, there's a mixture of the new gifs you mentioned with the older ones when running the local collections - The copy on the readme would also need to be updated too.

michael-wilkinson commented:
Is it possible to nudge this to get the ability to use API Keys brought into the main project?  I'm struggling to piece it together by looking at the pull request :( this is a great help and thanks for all the work so far!

DannyDainton commented:
There's parts of the readme that don't make sense after this change and although functional, it seems to work, it's going to tricky for folks to understand how it works, without another review of that file.

4nif commented:
@DannyDainton @wmedali  Is this PR still valid and can we use it for the Slack integration?

DannyDainton commented:
There's somethings that @wmedali needs to do before I merge any of it.

I've not really looked at this repo for a long time, I simply moved onto other things to be honest.

MichaelWWD commented:
Hello @DannyDainton , was wondering if we can now use this for the slack integration.
This is such a great help and thanks for all the work you've put into it
